legal-api/migrations/versions/540e5427636b_add_naics_tables.py

```diff
@@ -0,0 +1,118 @@
+from migrations.bulk_data.naics_elements_2017_v3_subset_testing_only import naics_elements_test_only
+
+# revision identifiers, used by Alembic.
+revision = '540e5427636b'
+down_revision = '94586877d4bc'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+
+    # Notes: bulk dictionary data for naics_structure and naics_element data used for the bulk inserts below were
+    # generated via a Jupyter notebook(/lear/tests/data/naics/create_alembic_import_data_from_naics_csvs.ipynb).
+    # The dictionary data are immported from migrations.bulk_data to make this migration file more readable and
+    # also easier to work with.
+    #
+    # Different data sets are used to overcome the slow performance of using the actual NAICS dataset for testing
+    # purposes.  As such when migrations are performed during test runs, only a small subset of data is used to load
+    # the test database for tests.
+
+    is_test_mode = current_app.config.get('TESTING', False)
+    print(f'is_test_mode: {is_test_mode}')
+
+    # naics_structures table
+    op.create_table('naics_structures',
+                    sa.Column('id', sa.Integer(), primary_key=True, nullable=False),
+                    sa.Column('naics_key', UUID(as_uuid=True), nullable=False, default=uuid.uuid4),
+                    sa.Column('level', sa.Integer(), nullable=False),
+                    sa.Column('hierarchical_structure', sa.String(length=25), nullable=False),
+                    sa.Column('code', sa.String(length=10), nullable=False),
+                    sa.Column('year', sa.Integer(), nullable=False),
+                    sa.Column('version', sa.Integer(), nullable=False),
+                    sa.Column('class_title', sa.String(length=150), nullable=False),
+                    sa.Column('superscript', sa.String(length=5), nullable=True),
+                    sa.Column('class_definition', sa.String(length=5100), nullable=False),
+                    sa.PrimaryKeyConstraint('id')
+                    )
+    op.create_index(op.f('ix_naics_structure_level'), 'naics_structures', ['level'], unique=False)
+    op.create_index(op.f('ix_naics_structure_code'), 'naics_structures', ['code'], unique=False)
+    op.create_index(op.f('ix_naics_structure_year'), 'naics_structures', ['year'], unique=False)
+    op.create_index(op.f('ix_naics_structure_version'), 'naics_structures', ['version'], unique=False)
+    op.create_index(op.f('ix_naics_structure_class_title'), 'naics_structures', ['class_title'], unique=False)
+    op.create_index(op.f('ix_naics_structure_class_defn'), 'naics_structures', ['class_definition'], unique=False)
+
+    meta = MetaData(bind=op.get_bind())
+    meta.reflect(only=('naics_structures',))
+    naics_structures_table = Table('naics_structures', meta)
+
+    naics_structures_to_insert = naics_structures_test_only if is_test_mode else naics_structures
+    op.bulk_insert(
```

Comment:
We should discuss if we want to change our deployment method.
So far , since we roll in structure changes with deployments, we haven't be doing data loads. Mostly because the load might take too long and the deployment fails.

Reply:
it takes about 50 seconds to data load locally right now. so it will probably deploy successfully in openshift environments…i think. but if you want i can yank the data loading stuff out and load it some other way.

Reply:
will leave as is for now as per discussion with Thor.  just need to keep in mind that we may need to revisit how we do larger data loads in the future.